include/tvm/runtime/packed_func.h

```diff
@@ -833,35 +859,17 @@ class TVMRetValue : public TVMPODValue_ {
   using TVMPODValue_::AsObjectRef;
   using TVMPODValue_::IsObjectRef;
 
-  TVMRetValue(const TVMRetValue& other) : TVMPODValue_() { this->Assign(other); }
   // conversion operators
-  operator std::string() const {
-    if (type_code_ == kTVMDataType) {
-      return DLDataType2String(operator DLDataType());
-    } else if (type_code_ == kTVMBytes) {
-      return *ptr<std::string>();
-    }
-    TVM_CHECK_TYPE_CODE(type_code_, kTVMStr);
-    return *ptr<std::string>();
-  }
-  operator DLDataType() const {
-    if (type_code_ == kTVMStr) {
-      return String2DLDataType(operator std::string());
-    }
-    TVM_CHECK_TYPE_CODE(type_code_, kTVMDataType);
-    return value_.v_type;
-  }
-  operator DataType() const { return DataType(operator DLDataType()); }
   template <typename FType>
   operator TypedPackedFunc<FType>() const {
     return TypedPackedFunc<FType>(operator PackedFunc());
   }
   // Assign operators
   TVMRetValue& operator=(TVMRetValue&& other) {
-    this->Clear();
-    value_ = other.value_;
-    type_code_ = other.type_code_;
-    other.type_code_ = kTVMNullptr;
+    std::swap(value_, other.value_);
+    std::swap(type_code_, other.type_code_);
+    std::swap(f_deleter_, other.f_deleter_);
+    std::swap(f_deleter_arg_, other.f_deleter_arg_);
```

Comment:
I think we are doing a bit too much special casing for this case. Instead, a simpler approach that might work is to leverage the Object system, which might align with our future directions of better FFI surrounding the object system